backend/src/database_pg.ts

```diff
@@ -200,12 +248,99 @@ export class Database {
         }
     }
 
-    public async close() {
-        try {
-            await this.pool.end();
-            Logger.logInfo('Closed the database connection.');
+
+
+
+    /**
+     **USERS TABLE 
+     *@method userExists
+     *@method addUser
+     *@method deleteUser 
+     *@method isAdmin
+     */
+
+    public async userExists(username:string){
+        const sql = 'SELECT COUNT(*) as count FROM users WHERE username = $1'; 
+        
+        try{
+
+            const res = await this.pool.query(sql, [username]);
+            Logger.logInfo(`User ${username} exists.`);
+            
+            return res.rows[0].count > 0;
+
+        } catch(err: any) {
+
+            Logger.logDebug(`User ${username} does not exist. `);
+            throw err;
+
+        }
+    }
+
+    public async addUser(username: string, is_admin: boolean, pw: string){
+        let pw_hash = await bcrypt.hash(pw, salt);
+
+        const sql = 'INSERT INTO users (username, is_admin, password) VALUES ($1, $2, $3)';
+
+        try{
+
+            const res = await this.pool.query(sql, [username, is_admin, pw_hash]);
+            Logger.logInfo(`User ${username} has been registered.`)
+
+        } catch(err: any){
+
+            Logger.logError("Error registering user: ", err);
+            throw err; 
+
+        }
+    }
+
+    //figure out how to wrap this so you have to validate the user deleting. extra function mayhaps?
+    public async deleteUser(username: string){
+        const sql = 'DELETE FROM users WHERE username = $1'; 
```

Comment:
We could try adding a rule that non-admin users cannot delete admins. The spec didn't specify how we needed to authenticate users and I'm worried that the automated testing may not handle it well. Another option would be to generate some sort of key, send it over API and ask the user to send the key back if they truly want to delete the user but again I'm not sure the autograder would handle this.

Reply:
handle in register endpoint before calling such function